### DISCUSSION TITLE
Starting MQTT on Neuron fails | MQTT Conversion

### DISCUSSION THREAD

**Discussion by YamatoYamm** [emqx/neuron#1814] [Q&A]
Hello !

I have 2 problems and I hope that someone could help me.

First problem is MQTT in Neuron, I followed steps from https://www.emqx.com/en/blog/bridging-opc-ua-data-to-mqtt-for-iiot but when I try to run MQTT even if it's ready, I receive Unknown error (4007) Adress Invalid(in log) and I don't receive any message in MQTTX. I've installed everything and I followed all steps.

Another question is: Can I have OPC UA over MQTT, complemented ? For example I need to have in my project OPC UA and MQTT but I don't want to convert OPC UA to MQTT, I need both same time. MQTT OSI level 4 and OPC UA higher level OSI.

**Comment by ZhouBox** [comment]
1st question: The log file shows a 4007 error indicating a connection problem between neuron and opcua. Please check the network configuration on site and the configuration of the neuron opcua node.

2nd question:Currently not supported.

**Comment by SuncNoob** [comment]
> Hello !
> 
> I have 2 problems and I hope that someone could help me.
> 
> First problem is MQTT in Neuron, I followed steps from https://www.emqx.com/en/blog/bridging-opc-ua-data-to-mqtt-for-iiot but when I try to run MQTT even if it's ready, I receive Unknown error (4007) Adress Invalid(in log) and I don't receive any message in MQTTX. I've installed everything and I followed all steps.
> 
> Another question is: Can I have OPC UA over MQTT, complemented ? For example I need to have in my project OPC UA and MQTT but I don't want to convert OPC UA to MQTT, I need both same time. MQTT OSI level 4 and OPC UA higher level OSI.

Hi Yamato,
If you are in Japan, we can arrange a meeting for a demo on NeuronEX and OPCUA. Both online and offline meetings are possible. If you are interested, please contact me at [cheng.sun@emqx.io](mailto:cheng.sun@emqx.io).
